src/routes/cspReport.js

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// In this router we implement the service to handle csp reports.
+
+import Router from '@koa/router';
+import body from 'koa-json-body';
+
+import { getLogger } from '../log';
+import { BadRequestError } from '../utils/errors';
+
+export function cspReportRoutes() {
+  const log = getLogger('cspReportRoutes');
+
+  const router = new Router();
+  router.post('/__cspreport__', body(), async ctx => {
+    if (!ctx.request.body) {
+      // Send a "Bad Request" error if the body could not be parsed.
+      throw new BadRequestError(`The body couldn't be parsed as JSON.`);
+    }
+
+    const cspReport = ctx.request.body['csp-report'];
+    if (!cspReport) {
+      // Send a "Bad Request" error as well if it doesn't look like a valid CSP report.
+      throw new BadRequestError(`The request misses a 'csp-report' field!`);
+    }
+
+    // Otherwise, just log the request
+    log.info('csp.report', cspReport);
```

Comment:
This is a general observation and thought, no action is needed here:

I find I don't know what the various log levels are for, or where they get sent.

For instance, I'm assuming that this gets stored in server logs, but I'm not really sure. I think it might be worth simplifying this mechanism. Logs are important for local debugging, and server debugging, but they are also a source of potential PII leakage. It would be nice to have more confidence on what's going on when a log function is run, and what type of information gets stored there.

Reply:
To be honest I'm not sure myself completely :)
I'm following the mozilla dockerflow recommendations, and they are that `info` logs are output in the console. That's all I know and I really don't know how they're handled later on... But I can find out!
Also I don't know how we act on these CSP reports later, the websec document doesn't say much about that, I'll also try to find out.

Reply:
I filed https://github.com/firefox-devtools/profiler-server/issues/35 so that I don't forget about finding out.